omod/src/main/webapp/fragments/patientdashboard/visits.gsp

```diff
@@ -57,12 +73,39 @@
             encounterCount = ${ param.encounterCount }; // This variable is defined in patientdashboard/patientDashboard.gsp
         <% } %>
         loadTemplates(visitId, ${ patient.id }, fromEncounter, encounterCount);
+
+        <%
+            def visits = patient.allVisitsUsingWrappers;
+            visits.eachWithIndex{ wrapper, idx ->
+                 if(wrapper.getVisit().visitId == param.visitId ||
+                    wrapper.getVisit().visitId == activeVisit.visit.id){
+                    if(wrapper.hasEncounters()){ %>
+                        jq('#i-toggle').show();
+                    <%}
+                 }
+            }
+        %>
     });
+
+    function toggle(){
+        jq('.collapse').removeClass('open');
+        jq(".view-details").click();
+        if(jq('#i-toggle').hasClass('icon-arrow-up')){
+            jq('#i-toggle').removeClass('icon-arrow-up');
+            jq('#i-toggle').addClass('icon-arrow-down');
+            jq('#i-toggle').attr('title', "${ui.message('coreapps.expandAll')}");
+        } else {
+            jq('#i-toggle').removeClass('icon-arrow-down');
+            jq('#i-toggle').addClass('icon-arrow-up');
+            jq('#i-toggle').attr('title', "${ui.message('coreapps.collapseAll')}");
+        }
+    }
 </script>
 
 <ul id="visits-list" class="left-menu">
 
-    <%  def visits = patient.allVisitsUsingWrappers
+    <%
+        visits = patient.allVisitsUsingWrappers
```

Comment:
Do you need to re-set the value of visits here? Can't you use the value from above?